pkg/hubble/observer/local_observer.go

```diff
@@ -450,8 +454,9 @@ func newRingReader(ring *container.Ring, req *observerpb.GetFlowsRequest, whitel
 	// In order to avoid buffering events, we have to rewind first to find the
 	// correct index, then create a new reader that starts from there
 	for i := ring.Len(); i > 0; i, idx = i-1, idx-1 {
-		e := reader.Previous()
-		if e == nil {
+		e, err := reader.Previous()
+		if err != nil {
+			idx++ // we went backward 1 too far
```

Comment:
> we went backward 1 too far

Unless the error is `io.EOF`, no?

Reply:
Kinda, I was not sure how to handle this properly. If `Previous` returns `io.EOF`, then something went horribly wrong. But yeah, we can check the error for `ErrConcurrentWrite`.